src/main/java/de/tum/in/www1/artemis/service/ModelingSubmissionService.java

```diff
@@ -93,12 +78,13 @@ public ModelingSubmission lockModelingSubmissionWithoutResult(ModelingExercise m
      *
      * @param modelingExercise the modeling exercise for which we want to get a modeling submission without result
      * @param correctionRound - the correction round we want our submission to have results for
-     * @param examMode flag to determine if test runs should be ignored. This should be set to true for exam exercises
+     * @param examMode flag to determine if test runs should be removed. This should be set to true for exam exercises
      * @return a modeling submission without any result
      */
-    public Optional<ModelingSubmission> getRandomModelingSubmissionEligibleForNewAssessment(ModelingExercise modelingExercise, boolean examMode, long correctionRound) {
+    private Optional<ModelingSubmission> getRandomModelingSubmissionEligibleForNewAssessment(ModelingExercise modelingExercise, boolean examMode, long correctionRound) {
         // if the diagram type is supported by Compass, ask Compass for optimal (i.e. most knowledge gain for automatic assessments) submissions to assess next
-        if (compassService.isSupported(modelingExercise)) {
+        // NOTE: compass only makes sense for the first correction round (i.e. correctionRound == 0)
+        if (compassService.isSupported(modelingExercise) && correctionRound == 0) {
```

Comment:
For future reference (not important that you change it here): Usually the "faster" condition is written first:
`correctionRound == 0 && compassService.isSupported(modelingExercise)`. When the round is not zero we don't need to call the method. (Currently the method with its performance overhead is called even if we throw the result away in the end because of the correction round being non-zero.)